src/components/CloudHeader/CloudHeader.js

```diff
@@ -0,0 +1,122 @@
+import CloudHeaderLogo from './CloudHeaderLogo';
+import CloudHeaderList from './CloudHeaderList';
+import CloudHeaderListItem from './CloudHeaderListItem';
+
+export default class CloudHeaderV2 extends React.Component {
+  state = {
+    isMenuOpen: false,
+  };
+
+  handleMenuClick = () => {
+    this.setState({
+      isMenuOpen: !this.state.isMenuOpen,
+    });
+  };
+
+  componentDidMount() {
+    this.portalNode = document.createElement('div');
+    document.body.appendChild(this.portalNode);
+  }
+
+  componentWillUnmount() {
+    this.portalNode.parentNode.removeChild(this.portalNode);
+  }
+
+  render() {
+    const {
+      className,
+      companyName,
+      productName,
+      logoHref,
+      renderMenu,
+      renderLogo,
+      renderSearch,
+      renderNotification,
+      renderApplications,
+      renderUser,
+      links,
+      ...other
+    } = this.props;
+    const { isMenuOpen } = this.state;
+    const cloudHeaderClasses = classNames('bx--cloud-header', className);
+
+    return (
+      <nav key="nav" className={cloudHeaderClasses} {...other}>
+        <CloudHeaderWrapper>
+          {renderMenu && <CloudHeaderMenu onClick={this.handleMenuClick} />}
+          <CloudHeaderLogo
+            className={!renderMenu && 'bx--cloud-header-brand--no-menu'}
+            companyName={companyName}
+            productName={productName}
+            href={logoHref}>
+            {renderLogo && renderLogo()}
+          </CloudHeaderLogo>
+          <CloudHeaderList>
+            {links &&
+              links.map(link => {
+                return (
+                  <CloudHeaderListItem href={link.href}>
+                    {link.linkText}
+                  </CloudHeaderListItem>
+                );
+              })}
+          </CloudHeaderList>
+        </CloudHeaderWrapper>
+        <CloudHeaderWrapper>
+          <CloudHeaderList>
+            {renderSearch && (
+              <CloudHeaderListItem onClick={renderSearch} isIcon>
+                <svg viewBox="0 0 16 16" fillRule="evenodd">
+                  <title>search</title>
+                  <path d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zm10 13.8L13.8 16l-3.6-3.6 2.2-2.2z" />
+                  <path d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z" />
+                </svg>
+              </CloudHeaderListItem>
+            )}
+            {renderNotification && (
+              <CloudHeaderListItem onClick={renderNotification} isIcon>
+                <svg width="16" height="16">
+                  <title>notifications</title>
+                  <path d="M9 1.11V0H7v1.11A5.022 5.022 0 0 0 3.1 4.9L1 14h5a2 2 0 0 0 4 0h5l-2.1-9.1A5.022 5.022 0 0 0 9 1.11z" />
+                </svg>
+              </CloudHeaderListItem>
+            )}
+            {renderApplications && (
+              <CloudHeaderListItem onClick={renderApplications} isIcon>
+                <svg width="16" height="16">
```

Comment:
Should we wrap up these `svg` elements as `UserIcon` or `ApplicationsIcon`, etc?